src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,303 @@
+            {
+                Thread.Sleep(14);
+            }
+            else if (this._mode.Equals(Sampling.UltraHighResolution))
+            {
+                Thread.Sleep(26);
+            }
+            else
+            {
+                Thread.Sleep(8);
+            }
+
+            int msb = Read8BitsFromRegister((byte)Register.PRESSUREDATA);            
+            int lsb = Read8BitsFromRegister((byte)Register.PRESSUREDATA + 1);            
+            int xlsb = Read8BitsFromRegister((byte)Register.PRESSUREDATA + 2);
+
+            int raw = ((msb << 16) + (lsb << 8) + xlsb) >> (8 - (byte)Sampling.Standard);
+            
+            return raw;            
+        }
+
+        /// <summary>
+        ///  Reads an 8 bit value from a register
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        internal byte Read8BitsFromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {                
+                _i2cDevice.WriteByte(register);
+                byte value = _i2cDevice.ReadByte();
+                return value;
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        ///  Reads a 16 bit value over I2C 
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value from register
+        /// </returns>
+        internal ushort Read16BitsFromRegister(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                Span<byte> bytes = stackalloc byte[2];
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(bytes);
+
+                return BinaryPrimitives.ReadUInt16LittleEndian(bytes);
+            }
+            else
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        ///  Reads a 16 bit value over I2C 
+        /// </summary>
+        /// <param name="register">
+        ///  Register to read from
+        /// </param>
+        /// <returns>
+        ///  Value (BigEndian) from register 
+        /// </returns>
+        internal ushort Read16BitsFromRegisterBE(byte register)
+        {
+            if (_communicationProtocol == CommunicationProtocol.I2c)
+            {
+                Span<byte> bytes = stackalloc byte[2];
+
+                _i2cDevice.WriteByte(register);
+                _i2cDevice.Read(bytes);
+
+                return BinaryPrimitives.ReadUInt16BigEndian(bytes);
+            }
+            else
+            {
+                throw new NotImplementedException();

```

Comment:
Maybe `Debug.Assert` that `_communicationProtocol == CommunicationProtocol.I2c` instead? With the current implementation we never hit this anyway.